helm/hpcc/templates/_helpers.tpl

```diff
@@ -45,19 +45,47 @@ Translate a port list to a comma-separated list
  {{- end -}}
 {{- end -}}
 
+{{- define "hpcc.getFirstPlaneForLabel" -}}
+{{- $root := .root -}}
+{{- $label := .label -}}
+{{- $storage := ($root.Values.storage | default dict) -}}
+{{- $planes := ($storage.planes | default list) -}}
+{{- $firstPlane := dict -}}
+{{- range $plane := $planes -}}
+{{- $labels := $plane.labels | default (list "data") -}}
+{{- if (has $label $labels) -}}
+{{- $_ := set $firstPlane "plane" $plane.name -}}
+{{- end -}}
+{{- end -}}
+{{- if $firstPlane.plane -}}
+{{- $firstPlane.plane -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "hpcc.hasPlaneForLabel" -}}
+{{- if (include "hpcc.getFirstPlaneForLabel" .) -}}
+true
+{{- end -}}
+{{- end -}}
+
 {{/*
 Get default data plane
 */}}
 {{- define "hpcc.getDefaultDataPlane" -}}
 {{- $storage := ($.Values.storage | default dict) -}}
-{{- $planes := ($storage.planes | default list) -}}
-{{- $firstPlane := dict -}}
-{{- range $plane := $planes -}}
- {{- if and (not $firstPlane.plane) (or (not $plane.labels) (has "data" $plane.labels)) }}
- {{- $_ := set $firstPlane "plane" $plane.name -}}
- {{- end -}}
+{{- $dataStorage := ($storage.dataStorage | default dict) -}}
+{{- $firstPlane := (include "hpcc.getFirstPlaneForLabel" (dict "root" $ "label" "data")) -}}
+{{- $dataStorage.plane | default $firstPlane | default "hpcc-data-plane" -}}
```

Comment:
all feels a little too complicated still in code/template/auto.

Perhaps would be simpler not to default to a implicit "hpcc-data-plane" if nothing defined,
but require at least one (can be ephemeral) and for default data planes (and same for other storage types) to be required.

The code here, assumptions, documentation and explanation would be simpler I think.

Reply:
I think that is a good goal - but we will break most existing values files if we make that change - so I would delay that until we are confident this method works (e.g., all examples and our own values file are moved over to defining planes).